src/core/meshcop/border_agent.cpp

```diff
@@ -776,13 +794,65 @@ void BorderAgent::Stop(void)
     Get<Tmf::SecureAgent>().Stop();
 
     mState        = kStateStopped;
+    mPskcChanged  = false;
     mUdpProxyPort = 0;
     LogInfo("Border Agent stopped");
 
 exit:
     return;
 }
 
+const char *BorderAgent::ReasonToString(RestartReason aReason)
+{
+    static const char *const kReasonStrings[] = {
+        "PskcChanged",               // (0) kRestartOnPskcChanged
+        "SecureSessionDisconnected", // (1) kRestartOnDisconnected
+        "EphemeralKeyRemoved",       // (2) kRestartOnEphemeralKeyRemoved
+
+    };
+
```

Comment:
Let's add checks like this to detect if enumerator values get changed in future.

```c++
static_assert(kRestartOnPskcChanged == 0, "kRestartOnPskcChanged value is incorrect");
static_assert(kRestartOnDisconnected == 1, "kRestartOnDisconnected value is incorrect");
static_assert(kRestartOnEphemeralKeyRemoved == 2, "kRestartOnEphemeralKeyRemoved value is incorrect");
 ```